src/helpers/build-routing-graph.js

```diff
@@ -0,0 +1,56 @@
+const graphlib = require("graphlib");
+const { isNil, get } = require("lodash");
+
+const setEdgeFromRoutingRule = (survey, nextPageId, destination, pageId) => {
+  switch (get(destination, "__typename")) {
+    case "LogicalDestination": {
+      const logicalDestination = get(destination, "logicalDestination");
+      if (logicalDestination === "NextPage") {
+        survey.setEdge(pageId, nextPageId);
+      }
+      break;
+    }
+    case "AbsoluteDestination": {
+      const absoluteDestinationId = get(destination, "absoluteDestination.id");
+      survey.setEdge(pageId, absoluteDestinationId);
+      break;
+    }
```

Comment:
Sure, for now I'll put the error as "Routing destination __typename not recognised", but this could be changed later if something more appropriate is made.